src/main/java/blusunrize/immersiveengineering/common/blocks/IEStairsBlock.java

```diff
@@ -9,36 +9,40 @@
 package blusunrize.immersiveengineering.common.blocks;
 
 import blusunrize.immersiveengineering.ImmersiveEngineering;
+import blusunrize.immersiveengineering.api.Lib;
 import blusunrize.immersiveengineering.common.IEContent;
+import net.minecraft.client.util.ITooltipFlag;
+import net.minecraft.item.BlockItem;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.block.*;
 import net.minecraft.block.BlockState;
-import net.minecraft.block.StairsBlock;
-import net.minecraft.entity.Entity;
-import net.minecraft.util.BlockRenderLayer;
-import net.minecraft.util.Direction;
-import net.minecraft.util.math.BlockPos;
-import net.minecraft.world.IBlockAccess;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.TranslationTextComponent;
+import net.minecraft.world.IBlockReader;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.api.distmarker.OnlyIn;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
 
 public class IEStairsBlock extends StairsBlock
 {
 	public boolean hasFlavour = false;
 	public boolean isFlammable = false;
-	public String name;
-	float explosionResistance;
-	BlockRenderLayer renderLayer = BlockRenderLayer.SOLID;
 
-	public IEStairsBlock(String name, BlockState state)
+	public IEStairsBlock(String name, BlockState state, Block.Properties properties)
 	{
-		super(state);
-		this.name = name;
-		this.setTranslationKey(ImmersiveEngineering.MODID+"."+name);
-		this.setCreativeTab(ImmersiveEngineering.itemGroup);
-		this.useNeighborBrightness = true;
-		this.explosionResistance = this.blockResistance/5f;
-//		ImmersiveEngineering.registerBlock(this, ItemBlockIEStairs.class, name);
+		super(state, properties);
+		setRegistryName(new ResourceLocation(ImmersiveEngineering.MODID, name));
 		IEContent.registeredIEBlocks.add(this);
-		IEContent.registeredIEItems.add(new ItemBlockIEStairs(this));
+		Item.Properties itemProps = new Item.Properties().group(ImmersiveEngineering.itemGroup);
+		IEContent.registeredIEItems.add(new BlockItem(this, itemProps));
 	}
 
+	//TODO: review required if this is needed.
```

Comment:
It doesn't seem to be used in 1.12, so feel free to remove it.

Reply:
Setter removed accordingly.